pkg/validator/ref_integrity.go

```diff
@@ -22,6 +22,11 @@ import (
 	"github.com/jmoiron/sqlx"
 )
 
+var (
+	ReferentialIntegrityErr = "referential integrity check failed at block %d, entry for %s not found"
+	EntryNotFoundErr        = "entry for %s not found"
+)
+
 // ValidateReferentialIntegrity validates referential integrity at the given height
 func ValidateReferentialIntegrity(db *sqlx.DB, blockNumber uint64) error {
```

Comment:
Yeah exactly

Reply:
Actually perhaps this wouldn't be desirable, since if an early validation fails we don't need to run the query for the subsequent ones... or maybe we do, so that we can report all the broken references not just the first we run into??

Reply:
Ultimately since these are read-only transactions, it shouldn't impact the overall failure behavior much. The code currently short-circuits on any failure so we'll still only see the first bad reference at a block, so we'd need to change the error handling first.

But, this would give us isolation so we know any failed refs exist at a singular point, which would increase the value of gathering multiple errors, if we want to. I think it's a good idea.

Reply:
@roysc let's go ahead and do this, thanks!

Reply:
Done. So now the only problem with this is the tests keep timing out, since the service just doesn't catch up to head. I've increased the timeout in the tests but something is still timing out at 10m, so I need to investigate that, and I might increase the slot interval if it keeps happening.

Reply:
Finally resolved this, there was a bug in handling the delay when waiting for the chain to catch up, which was making it take forever. I've fixed that and improved the logging for that case, and I think this is good to go now - will merge it once the latest commit passes CI.